logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/AuthenticationScope.kt

```diff
@@ -77,15 +46,29 @@ class AuthenticationScope(
     val validateUserHandleUseCase: ValidateUserHandleUseCase get() = ValidateUserHandleUseCaseImpl()
     val validatePasswordUseCase: ValidatePasswordUseCase get() = ValidatePasswordUseCaseImpl()
 
-    val addAuthenticatedAccount: AddAuthenticatedUserUseCase get() = AddAuthenticatedUserUseCase(sessionRepository)
-    val login: LoginUseCase get() = LoginUseCaseImpl(loginRepository, validateEmailUseCase, validateUserHandleUseCase)
-    val getSessions: GetSessionsUseCase get() = GetSessionsUseCase(sessionRepository)
-    val getServerConfig: GetServerConfigUseCase get() = GetServerConfigUseCase(serverConfigRepository)
-    val session: SessionScope get() = SessionScope(sessionRepository)
-    val register: RegisterScope get() = RegisterScope(registerAccountRepository)
-    val ssoLoginScope: SSOLoginScope get() = SSOLoginScope(ssoLoginRepository, sessionMapper)
-    val saveNotificationToken: SaveNotificationTokenUseCase get() = SaveNotificationTokenUseCase(notificationTokenRepository)
-    val enableLogging: EnableLoggingUseCase get() = EnableLoggingUseCaseImpl(userConfigRepository)
-    val isLoggingEnabled: IsLoggingEnabledUseCase get() = IsLoggingEnabledUseCaseImpl(userConfigRepository)
-    val buildConfigs: GetBuildConfigsUseCase get() = GetBuildConfigsUseCaseImpl(kaliumConfigs)
+    val login: LoginUseCase get() = LoginUseCaseImpl(loginRepository, validateEmailUseCase, validateUserHandleUseCase, backendLinks)
+    val register: RegisterScope get() = RegisterScope(registerAccountRepository, backendLinks)
+    val ssoLoginScope: SSOLoginScope get() = SSOLoginScope(ssoLoginRepository, backendLinks, globalScope.serverConfigRepository)
+}
+
+
+class ServerMetaDataManagerImpl internal constructor(
+    private val serverConfigRepository: ServerConfigRepository,
+    private val serverConfigMapper: ServerConfigMapper = MapperProvider.serverConfigMapper()
+) : ServerMetaDataManager {
+
+    override fun getLocalMetaData(backendLinks: ServerConfigDTO.Links): ServerConfigDTO? =
+        serverConfigRepository.configByLinks(serverConfigMapper.fromDTO(backendLinks)).nullableFold({
+            null
+        }, {
+            serverConfigMapper.toDTO(it)
+        })
+
+    override fun storeBackend(links: ServerConfigDTO.Links, metaData: ServerConfigDTO.MetaData): ServerConfigDTO? {
```

Comment:
it's a little peaky, but I think we can choose a better name for this function! specially when we call storeConfig! it's a little bit confusing.

Reply:
renamed to `storeServerConfig`